charms/worker/k8s/src/token_distributor.py

```diff
@@ -48,6 +49,129 @@ class ClusterTokenType(Enum):
     NONE = ""
 
 
+class TokenManager:
```

Comment:
Our token strategies dict was getting too complicated.  By making a Base Class ... we can safely describe how to create different token strategies